tutorials/23_MultiHop_QA_With_Agents.ipynb

```diff
@@ -0,0 +1,5680 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "h8gcxwg4sAXd"
+      },
+      "source": [
+        "# Tutorial: Multihop Question Answering with Agents\n",
+        "\n",
+        "- **Level**: Advanced\n",
+        "- **Time to complete**: 10 minutes\n",
+        "- **Nodes Used**: `Agent`, `PromptNode`, `InMemoryDocumentStore`, `FARMReader` and `ExtractiveQAPipeline`\n",
+        "- **Goal**: After completing this tutorial, you will have learned how to make use of Agents to build a multi-hop question answering system that uses an `ExtractiveQAPipeline` as a tool\n",
+        "- **Prerequisites**: An OpenAI API Key\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "collapsed": false,
+        "id": "ifCja5PasAXf"
+      },
+      "source": [
+        "## Overview\n",
+        "\n",
+        "The Agent class makes use of a large language model to take decisions and come up with the best next course of action. You are able to provide the `Agent` with a set of `Tools` that it is able to use to reach a result. \n",
+        "\n",
+        "In this tutorial, we will let the Agent use a commonly used Haystack component, the `ExtractiveQAPipeline` to answer question. At each iteration, the agent will decide what tool to use. Based on the result returned by a tool, the Agent has two options: It will either decide to select a tool again and do another iteration, or it will decide that it has reached a conclusion and return the final answer.\n"
```

Comment:
```suggestion
        "In this tutorial, we will let the Agent use a commonly used Haystack component, the `ExtractiveQAPipeline`, to answer questions. At each iteration, the agent will decide what tool to use. Based on the result returned by a tool, the Agent has two options: It will either decide to select a tool again and do another iteration, or it will decide that it has reached a conclusion and return the final answer.\n"
```

Reply:
is the agent going to get another tool? because here ew say it's going to get extractive pipeline and then we talk it will choose which tool to use but itlooks like it's going to have just one tool

Reply:
The agent _can_ get another tool. In this tutorial for simplicity it's just one. In a next tutorial, I'm thinking to create and add another tool yes.

Reply:
yes, but in this tutorial it only gets one tool. And here you're describing this tutorial ("in this tutorial.."), not the agent in general. I just think it may be a bit confusing. You could say that in general the agent can get multiple tools but in this tutorial we'll show just one not to overcomplicate things.

Reply:
I changed this paragraph structure quite a bit..